### ISSUE TITLE
Privacy-preserving web analytics

### ISSUE THREAD

**Issue by sandman** [welpo/tabi#172]
# Feature Request

## Summary
Enable free (ideally) privacy-preserving self-hosted web analytics for Tabi site.

## Motivation
Web analytics helps to understand site visit patterns and is useful for SEO, tailoring content etc. 

## Detailed Description
Integrate [Plausible](https://plausible.io/) as a first option within Tabi. Implementation details are TBD.

## Additional Context
N.A.

**Comment by donovanglover** [comment]
[Umami](https://umami.is/) is also an option. I've used it for a number of years.

**Comment by Jieiku** [comment]
I like Matomo because it supports parsing nginx logs while still supporting using javascript to collect the data. Not everyone has access to nginx logs, but if you do then it is a great way to do some basic analytics. One of the benefit to parsing logs instead of loading js is that it is faster for your website, and also it cannot be blocked by a browser plugin.

uBlock origin for example will block the majority of javascript based analytics trackers.

Some of the other tools may also support parsing logs, I have not looked at them yet.

**Comment by welpo** [comment]
Thanks for all the valuable input, @sandman, @donovanglover, and @Jieiku!

I've just created the branch [`feat/analytics`](https://github.com/welpo/tabi/tree/feat/analytics) that adds initial support for [Plausible](https://github.com/plausible/analytics), [Umami](https://github.com/umami-software/umami), and [GoatCounter](https://github.com/arp242/goatcounter).

I've only tested non-self-hosted Umami. If any of you would be willing to test the Plausible or GoatCounter integrations—or a self-hosted setup—that would be very much appreciated. The Content Security Policy (CSP) has also been updated to accommodate these services.

To set them up, check out the `config.toml` comments:

```toml
[extra.analytics]
# Specify which analytics service you want to use.
# Supported options: ["goatcounter", "umami", "plausible"]
service = "umami"

# Unique identifier for tracking.
# For GoatCounter, this is the code you choose during signup.
# For Umami, this is the website ID.
# For Plausible, this is the domain name (e.g. "example.com").
# Note: Leave this field empty if you're self-hosting.
id = "yourID"

# Optional: Specify the URL for self-hosted analytics instances.
# For GoatCounter: Base URL like "https://stats.example.com"
# For Umami: Base URL like "https://umami.example.com"
# For Plausible: Base URL like "https://plausible.example.com"
# Leave this field empty if you're using the service's default hosting.
self_hosted_url = ""
```

Looking forward to your feedback.

**Comment by welpo** [comment]
I've merged the changes in #193. If anyone encounters a problem with analytics, please open a new issue.

Thanks!
